starter-code/src/viking.js

```diff
@@ -1,11 +1,94 @@
 // Soldier
-function Soldier() {}
+function Soldier(health,strength) {
+    this.health=health;
+    this.strength=strength;
+}
+Soldier.prototype.attack=function(){
+    return this.strength;
+}
+Soldier.prototype.receiveDamage=function(theDamage){
+    this.health=this.health-theDamage;
+    return
+}
 
 // Viking
-function Viking() {}
+Viking.prototype = Object.create(Soldier.prototype);
 
+function Viking (name, health, strength) { 
+    this.name = name; 
+    this.health = health;
+    this.strength = strength;  
+}
+Viking.prototype.attack=function(){
+    return this.strength;
+}
+Viking.prototype.receiveDamage=function(theDamage){
+    this.health=this.health-theDamage;
+    if (this.health > 0) {
+        return this.name + " has received " + theDamage + " points of damage";
+    }
+    return this.name + " has died in act of combat"; 
+}
+Viking.prototype.battleCry=function() {
+    return "Odin Owns You All!"
+
+}
 // Saxon
-function Saxon() {}
+Saxon.prototype = Object.create(Soldier.prototype);
+
+function Saxon(health,strength){
+    this.strength=strength
+    this.health=health
+}
+
+Saxon.prototype.attack=function(){
+    return this.strength;
+}
+Saxon.prototype.receiveDamage=function(theDamage){
+    this.health=this.health-theDamage;
+    if (this.health > 0) {
+        return  "A Saxon has received " + theDamage + " points of damage";
+    }
+    return  "A Saxon has died in combat"; 
+}
 
 // War
-function War() {}
+function randomSelector(array){
+    return array[Math.floor(Math.random()*array.length)];
+     }
+
+function War() {
+    this.vikingArmy=[];
+    this.saxonArmy=[];
+    this.addViking=function(Viking){
+        this.vikingArmy.push(Viking);
+        return;
+    }
+    this.addSaxon=function(Saxon){
+        this.saxonArmy.push(Saxon);
+        return;
+    }
+    this.vikingAttack=function() {
```

Comment:
Estaban bien encaminados, pero les faltaba un poco. Solucion:

    this.vikingAttack = function(){
        var randomSaxon = Math.floor(Math.random()*this.saxonArmy.length);
        var randomViking = Math.floor(Math.random()*this.vikingArmy.length);
        this.saxonArmy[randomSaxon].receiveDamage(this.vikingArmy[randomViking].strength);
        if (this.saxonArmy[randomSaxon].health<=0) {
            this.saxonArmy.splice(randomSaxon,1);
            return "A Saxon has died in combat";
        } else {
            return "A Saxon has received " + this.vikingArmy[randomViking].strength + " points of damage";
        }
    };